sample-compose/data/src/main/java/co/nimblehq/sample/compose/data/repository/AppPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.sample.compose.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.sample.compose.domain.repository.AppPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class AppPreferencesRepositoryImpl @Inject constructor(
+    private val appPreferencesDataStore: DataStore<Preferences>
+) : AppPreferencesRepository {
+
+    private object PreferencesKeys {
+        val FIRST_TIME_LAUNCH = booleanPreferencesKey("FIRST_TIME_LAUNCH")
+    }
+
+    override fun isFirstTimeLaunchPreferences(): Flow<Boolean> = flow {
+        val firstLaunchPreferences = appPreferencesDataStore.data
+            .catch { exception ->
+                if (exception is IOException) {
+                    Log.e(
+                        AppPreferencesRepositoryImpl::class.simpleName,
+                        "Error reading preferences.",
+                        exception
+                    )
+                    emit(emptyPreferences())
```

Comment:
@chornerman I answered [here](https://github.com/nimblehq/android-templates/pull/389#discussion_r1088821962). I think handling the `IOException` is good to follow since it is what Google suggested. And the way we are handling it is just our sample to do, right? So in this case, I wanna create the `Empty Preference`, which means it will return the `default` value to the view. In this way, we will not see the `IOException` on the view and will see `This is the first time` instead. 
Thinking of the real use case, users don't need to know if there is something wrong with the device. Imagine we got exactly `IOException`, then what will we handle further? Just throw it and show a toast message, and the user will be blocked? I think we shouldn't handle it like that. Instead, returning an `Empty preference with default value` can resolve this. So it will not block users and will continue to flow smoothly.
cc @nimblehq/android-chapter

Reply:
> Thinking of the real use case, users don't need to know if there is something wrong with the device. Imagine we got exactly IOException, then what will we handle further? Just throw it and show a toast message, and the user will be blocked?

@lydiasama, I agree with you on this point, So that should we still `combine` the source of `getModelsUseCase()` and `getModelsUseCase()` in `VM` as it will be block when one of the sources returns an error? What do you think? 🙏

Reply:
@Wadeewee My point in combining them all together is I'd like to demonstrate how we handle the chain use cases in a real project. When working on a client project, we usually combine all use cases to ensure the result will not be the `race-condition` case. 

Besides, I think it's acceptable to do it in this case. The worst case of the DataStore is the `IOException` that was covered with `Empty preferences with the default value`. So I don't think there will be another exception returns from `getFirstTimeLaunchPreferencesUseCase`. After testing several times, I didn't see the exception at all.